security/pkg/nodeagent/caclient/client.go

```diff
@@ -45,13 +45,12 @@ type configMap interface {
 }
 
 // NewCAClient create an CA client.
-func NewCAClient(endpoint, caProviderName string, tlsFlag bool, tlsRootCert []byte, vaultAddr, vaultRole,
-	vaultAuthPath, vaultSignCsrPath string) (caClientInterface.Client, error) {
+func NewCAClient(endpoint, caProviderName string, tlsFlag bool, config string) (caClientInterface.Client, error) {
```

Comment:
I am trying to see where this is used, and as far as I can tell its used only by:

* NewSecretFetcher, used in server_test.go
* client_test.go

So this is basically just test code any doesn't really matter?

am I missing where its actually used?

Reply:
But regardless, I recommend a struct as well. If we want the user to pass 1 config, just make them pass the yaml for that struct.For example
```yaml
env:
- name: VAULT_CONFIG
  value: |
    jwtToken: foo
    something: bar
```

Reply:
Thanks for the comment John.

This is the first PR to enable Vault-Istio integration via istiod. The latter is used as an RA and Vault is the backend CA. I'll update istiod in a followup PR. This is a feature for our long-term goal of supporting using istiod as an RA. Of course, it unblocks Amex :)

I have updated the ENV variable loading - to make it loaded inside vault/client.go. Your config looks pretty good... Can you show me an ENV with that format?